packages/adblocker/src/engine/engine.ts

```diff
@@ -229,6 +229,79 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
     return engine as InstanceType<T>;
   }
 
+  public static merge<T extends FilterEngine>(this: new (...args: any[]) => T, engines: T[]): T {
+    if (!engines || engines.length < 2) {
+      throw new Error('merging engines requires at least two engines');
+    }
+
+    const config = engines[0].config;
+    const lists = engines[0].lists;
+    const engineVersion = engines[0].engineVersion;
+
+    const networkFilters: Map<number, NetworkFilter> = new Map();
+    const cosmeticFilters: Map<number, CosmeticFilter> = new Map();
+    const preprocessors: Preprocessor[] = [];
+
+    const configKeysMustMatch: Exclude<keyof Config, 'serialize' | 'getSerializedSize'>[] = [
+      'debug',
+      'enableCompression',
+      'enableOptimizations',
+    ];
+
+    for (const engine of engines) {
+      // Validate the config
+      for (const configKey of configKeysMustMatch) {
+        if (config[configKey] !== engine.config[configKey]) {
+          throw new Error(`config "${configKey}" of all merged engines must be the same`);
+        }
+      }
+
+      if (engineVersion !== engine.engineVersion) {
+        throw new Error(
+          `engine version mismatch during merge, expected ${engineVersion} but got ${engine.engineVersion}`,
+        );
+      }
+
+      const filters = engine.getFilters();
+
+      if (config.loadNetworkFilters === true) {
+        for (const networkFilter of filters.networkFilters) {
+          networkFilters.set(networkFilter.getId(), networkFilter);
+        }
+      }
```

Comment:
I'm not sure we save anything by checking before since the list will be `[]` so it's a no-op. Also, in the current state it would be allowed for some engines to have `loadNetworkFilters` to `true` and others to have it to `false`, which would break this "optimization". I'd be in favor of removing the check.

Reply:
Updated in [`05c8b8f` (#4138)](https://github.com/ghostery/adblocker/pull/4138/commits/05c8b8f2b9977881be140127407d51316d2ab727).